src/custom/hooks/useDebounceWithForceUpdate.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react'
+
+// modified from https://usehooks.com/useDebounce/
+export default function useDebounceWithForceUpdate<T>(
+  value: T,
+  delay: number,
+  forceUpdateRef?: string | number | boolean
+): T {
+  const [lastForceUpdateRef, setLastForceUpdateRef] = useState(forceUpdateRef)
+  const [debouncedValue, setDebouncedValue] = useState<T>(value)
```

Comment:
Why 2 states? if we have only one value, we update it eagerly when ref changes, and with a delay when it doesn't